### PR TITLE
Refactor package-deps-hash for improved testability

### PR DESCRIPTION
**Pull request by nickpape-msft**

### CODE REVIEW

File: libraries/package-deps-hash/src/getPackageDeps.ts [hunk 1]
```diff
@@ -1,86 +1,148 @@
...
+}
+
+/**
+ * Executes "git ls-tree" in a folder
+ */
+export function gitLsTree(path: string): string {
+  return child_process.execSync(
+    `git ls-tree HEAD -r`,
+    { cwd: path }).toString();
+}
```
Comment (pgonzal) [inline libraries/package-deps-hash/src/getPackageDeps.ts]: It's weird that we are executing bare shell commands without knowing where the PATH points.  Maybe out of scope for this PR?

File: libraries/package-deps-hash/src/getPackageDeps.ts [hunk 2]
```diff
@@ -1,86 +1,148 @@
...
+  if (output) {
+    // Note: The output of git ls-tree uses \n newlines regardless of OS.
+    output.split('\n').forEach(line => {
+
+      // A line is expected to look like:
+      // 100644 blob 3451bccdc831cb43d7a70ed8e628dcf9c7f888c8    src/typings/tsd.d.ts
+
+      if (line) {
+        // Take everything after the "100644 blob", which is just the hash and filename
+        const [hash, filename]: string[] = line.substr(line.indexOf('blob ') + 5).split('\t');
```
Comment (pgonzal) [inline libraries/package-deps-hash/src/getPackageDeps.ts]: >line [](start = 43, length = 4)

(optional) Would be better to use  a RegExp

File: libraries/package-deps-hash/src/getPackageDeps.ts [hunk 3]
```diff
@@ -1,86 +1,148 @@
 import * as child_process from 'child_process';
 import { IPackageDeps } from './IPackageDeps';
 
-const PROCESS_OUTPUT_DELIMITER: string = '///~X~X~X~X~X~X~///';
+export type GitStatusChangeType = 'D' | 'M' | 'A';
 
-/** @public */
+/**
+ * Parses the output of the "git ls-tree" command
```
Comment (pgonzal) [inline libraries/package-deps-hash/src/getPackageDeps.ts]: >Parses [](start = 3, length = 6)

The return value would be pretty useful to document

**Comment by pgonzal** [comment]
:shipit:

File: libraries/package-deps-hash/src/getPackageDeps.ts (hunk 2 above)
Comment (nickpape-msft) [inline libraries/package-deps-hash/src/getPackageDeps.ts]: That's in a different PR, which was the one I was trying to merge in the first place.

**Pull request merged**